website/en/docs/lang/common-mistakes.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: guide
+---
+
+Often, Flow will raise errors that you would consider wrong. Most of these issues are actually explained in the presentation of the [type system](https://flow.org/en/docs/lang/) and more specifically in the [soundness and completeness section](https://flow.org/en/docs/lang/types-and-expressions/#toc-soundness-and-completeness).
+Several of them are the consequence of careful tradeoffs made by the team, where you have to balance false positives and false negatives (see the above mentionned section).
+
+### Common user errors <a class="toc" id="toc-common-user-errors" href="#toc-common-user-errors"></a>
+
+#### Refinement invalidation
```

Comment:
I checked that `foo.bar` is not `null`, but Flow still thinks it is. Why does this happen?